<a href="https://colab.research.google.com/github/jeguns/EP7173/blob/main/Unidad%2002/02_Procedimientos_b%C3%A1sicos_de_preprocesamiento_de_datos.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Procedimientos básicos de preprocesamiento de datos**

En esta oportunidad, trabajaremos con tablas de datos provenientes de la Encuesta Nacional de Hogares (ENAHO) del primer trimestre del año 2024. En específico, utilizaremos los módulos 100 (Características de la Vivienda y del Hogar) y 200 (Características de los Miembros del Hogar).

Fuente de datos: https://proyectos.inei.gob.pe/microdatos/

## **Pregunta 1**

Leer cada uno de los archivos indicados

In [1]:
import pandas as pd

In [4]:
pd.set_option('display.max_columns', None)

In [2]:
datos1=pd.read_csv('Enaho01-2024-100.csv', encoding='latin-1')
datos2=pd.read_csv('Enaho01-2024-200.csv', encoding='latin-1')

In [5]:
datos1

AÑO  MES  CONGLOME  VIVIENDA  HOGAR  UBIGEO  DOMINIO  ESTRATO  \
0      2024    1     15009        10     11   10101        4        4   
1      2024    1     15009        22     11   10101        4        4   
2      2024    1     15009        47     11   10101        4        4   
3      2024    1     15009        59     11   10101        4        4   
4      2024    1     15009        71     11   10101        4        4   
...     ...  ...       ...       ...    ...     ...      ...      ...   
11221  2024    3     20692        40     11  150801        2        2   
11222  2024    3     20692        55     11  150801        2        2   
11223  2024    3     20692        70     11  150801        2        2   
11224  2024    3     20692        84     11  150801        2        2   
11225  2024    3     20692        99     11  150801        2        2   

       PERIODO  TIPENC    FECENT  RESULT PANEL P22     P23 P24A P24B P25$1  \
0            1       1  20240123       4                 .                   
1            1       1  20240124       2         2       .    1    3     1   
2            1       1  20240130       1         2       .    2    3     1   
3            1       1  20240117       3                 .                   
4            1       1  20240128       1         2       .    2    1     0   
...        ...     ...       ...     ...   ...  ..     ...  ...  ...   ...   
11221        4       1  20240330       2         2       .    4    3     1   
11222        4       1  20240329       5                 .                   
11223        4       1  20240329       1         2       .    1    1     0   
11224        4       1  20240329       1         2       .    4    3     0   
11225        4       1  20240331       1         2       .    1    2     0   

      P25$2 P25$3 P25$4 P25$5 P101 P102 P103 P103A P104 P104A P104B1 P104B2  \
0                                                                             
1         0     1     1     0    1    1    3     1    6     4      1      1   
2         0     1     1     0    1    1    3     1    5     3      2      2   
3                                                                             
4         1     1     1     0    3    3    5     4    1     1      3      3   
...     ...   ...   ...   ...  ...  ...  ...   ...  ...   ...    ...    ...   
11221     0     0     1     0    1    7    5     4    4     3      2      2   
11222                                                                         
11223     1     0     1     0    1    1    5     1    2     1      2      2   
11224     1     0     1     0    1    7    6     4    3     2      2      2   
11225     1     0     1     0    1    1    5     1    5     4      2      2   

      P105A P105B  P106 P106A P106B P107B1 P107C11 P107C12 P107C13 P107C14  \
0                                                                            
1         2        3000     1     1      2                                   
2         6         750                  2                                   
3                                                                            
4         6         400                  2                                   
...     ...   ...   ...   ...   ...    ...     ...     ...     ...     ...   
11221     2         100     2            2                                   
11222                                                                        
11223     2          50     1     1      2                                   
11224     2         150     2            2                                   
11225     2         200     2            2                                   

      P107C16 P107C17 P107C18 P107C19 P107C110 P107D1 P107B2 P107C21 P107C22  \
0                                                                              
1                                                          2                   
2                                                          2        

In [6]:
datos2

AÑO  MES  CONGLOME  VIVIENDA  HOGAR  UBIGEO  DOMINIO  ESTRATO  \
0      2024    1     15009        22     11   10101        4        4   
1      2024    1     15009        22     11   10101        4        4   
2      2024    1     15009        22     11   10101        4        4   
3      2024    1     15009        22     11   10101        4        4   
4      2024    1     15009        47     11   10101        4        4   
...     ...  ...       ...       ...    ...     ...      ...      ...   
28158  2024    3     20692        84     11  150801        2        2   
28159  2024    3     20692        84     11  150801        2        2   
28160  2024    3     20692        99     11  150801        2        2   
28161  2024    3     20692        99     11  150801        2        2   
28162  2024    3     20692        99     11  150801        2        2   

       CODPERSO              P201P  P203 P203A P203B P204 P205 P206 P207  \
0             1  20240150090221101     1     1     1    1    2         1   
1             2  20240150090221102     2     1     2    1    2         2   
2             3  20240150090221103     3     1     3    1    2         1   
3             4  20240150090221104     3     1     3    1    2         2   
4             1  20240150090471101     1     1     1    1    2         2   
...         ...                ...   ...   ...   ...  ...  ...  ...  ...   
28158         4  20240206920841104     3     1     3    1    2         2   
28159         5  20240206920841105     3     1     3    1    2         1   
28160         1  20240206920991101     1     1     1    1    2         2   
28161         2  20240206920991102     3     1     3    1    2         2   
28162         3  20240206920991103     3     1     3    1    2         1   

      P208A P208B P209 P210 P211A P211D P212 P213 P214 P215 P216 P217  \
0        52          2                     1    1    1                  
1        47          2                     2    2    2                  
2        18          6                     3    3    3                  
3        11               2    10          4    4    0                  
4        39          5                     1    1    1                  
...     ...   ...  ...  ...   ...   ...  ...  ...  ...  ...  ...  ...   
28158    25          6                     4    4    4                  
28159    23          6                     5    5    5                  
28160    60          3                     1    1    1                  
28161    39          6                     2    2    2                  
28162    28          6                     3    3    3                  

      OCUPAC_R3 OCUPAC_R4 RAMA_R3 RAMA_R4 CODTAREA CODTIEMPO T211  TICUEST01  \
0                                                                          2   
1                                                                          2   
2                                                                          2   
3                                                              11          2   
4                                                                          2   
...         ...       ...     ...     ...      ...       ...  ...        ...   
28158                                                                      2   
28159                                                                      2   
28160                                                                      2   
28161                                                                      2   
28162                                                                      2   

                 FACPOB  
0       339.76591648557  
1      295.776571049962  
2      392.547179621175  
3      415.498038496145  
4      400.218362878378  
...                 ...  
28158  1368.68004865069  
28159  1342.44588138718  
28160  1011.50654043682  
28161  1368.68004865069  
28162  1616.28961792146  

[28163 rows x 38 columns]

## **Pregunta 2**

Seleccione las variables que se mencionan a continuación:

Para ambos módulos, considerar las siguientes variables:

- CONGLOME: Número de conglomerado

- VIVIENDA: Número de selección de vivienda

- HOGAR: Número secuencial del Hogar

- UBIGEO: Ubicación geográfica

En particular, del módulo 100 se requerirá:

- FECENT: Fecha de Resultado final de la encuesta (Mes/Día/Año)

- P101: Tipo de vivienda
    1. Casa independiente
    2. Departamento en edificio
    3. Vivienda en quinta
    4. Vivienda en casa de vecindad(callejón solar o corralón)
    5. Choza o cabaña
    6. Vivienda improvisada
    7. Local no destinado para habitación humana
    8. Otro

- P104: ¿Cuántas habitaciones en total tiene la vivienda, sin contar el baño,la cocina, los pasadizos, ni el garage?

- P106: ¿Cuánto cree que le pagarían de alquiler mensual (en S/.) ?

- P106A: ¿Esta vivienda tiene titulo de propiedad?
    1. Si
    2. No

- P110: El agua que utilizan en el hogar ¿Procede principalmente de:
    1. Red pública, dentro de la vivienda
    2. Red pública, fuera de la vivienda pero dentro del edificio
    3. Pilón o pileta de uso público
    4. Camión-cisterna u otro similar
    5. Pozo (agua subterránea)
    6. Manantial o puquio
    7. Otra
    8. Río, acequia, lago, laguna

- P1141: Su hogar tiene: Teléfono (fijo):
    0. Pase
    1. Teléfono(fijo)

- P1142: Su hogar tiene: Teléfono Celular:
    0. Pase
    1. Teléfono Celular

- P1143: Su hogar tiene: Conexion a TV por cable o satelital
    0. Pase
    1. Conexión a TV por cable o satelital

- P1144: Su hogar tiene: Conexion a Internet
    0. Pase
    1. Conexión a Internet

Por otro lado, del módulo 200 se requerirá:

- CODPERSO: Número de orden de la persona

- P203: ¿Cuál es la relación de parentesco con el jefe(a) del hogar?
    0. Panel
    1. Jefe/Jefa
    2. Esposo(a)/compañero(a)
    3. Hijo(a)/Hijastro(a)
    4. Yerno/Nuera
    5. Nieto(a)
    6. Padres/Suegros
    7. Otros parientes
    8. Trabajador Hogar
    9. Pensionista
    10. Otros no parientes
    11. Hermano(a)

- P207: Sexo
    1. Hombre
    2. Mujer

- P208A: ¿Qúe edad tiene en años cumplidos ? (En años)

- P209: ¿Cuál es su estado civil o conyugal?
    1. Conviviente
    2. Casado(a)
    3. Viudo(a)
    4. Divorciado(a)
    5. Separado(a)
    6. Soltero(a)

- TICUEST01: Origen de cuestionario
    1. Cuestionario en hojas
    2. Cuestionario en Tablet


In [7]:
datos1 = datos1[['CONGLOME','VIVIENDA','HOGAR','UBIGEO','FECENT','P101','P104','P106','P106A','P110','P1141','P1142','P1143','P1144']]
datos1

CONGLOME  VIVIENDA  HOGAR  UBIGEO    FECENT P101 P104  P106 P106A P110  \
0         15009        10     11   10101  20240123                              
1         15009        22     11   10101  20240124    1    6  3000     1    1   
2         15009        47     11   10101  20240130    1    5   750          1   
3         15009        59     11   10101  20240117                              
4         15009        71     11   10101  20240128    3    1   400          1   
...         ...       ...    ...     ...       ...  ...  ...   ...   ...  ...   
11221     20692        40     11  150801  20240330    1    4   100     2    6   
11222     20692        55     11  150801  20240329                              
11223     20692        70     11  150801  20240329    1    2    50     1    8   
11224     20692        84     11  150801  20240329    1    3   150     2    4   
11225     20692        99     11  150801  20240331    1    5   200     2    4   

      P1141 P1142 P1143 P1144  
0                              
1         0     1     1     1  
2         1     1     1     1  
3                              
4         0     1     1     1  
...     ...   ...   ...   ...  
11221     0     1     1     1  
11222                          
11223     0     1     1     1  
11224     0     1     1     1  
11225     0     1     1     1  

[11226 rows x 14 columns]

In [8]:
datos2 = datos2[['CONGLOME','VIVIENDA','HOGAR','UBIGEO','CODPERSO','P203','P207','P208A','P209','TICUEST01']]
datos2

CONGLOME  VIVIENDA  HOGAR  UBIGEO  CODPERSO  P203 P207 P208A P209  \
0         15009        22     11   10101         1     1    1    52    2   
1         15009        22     11   10101         2     2    2    47    2   
2         15009        22     11   10101         3     3    1    18    6   
3         15009        22     11   10101         4     3    2    11        
4         15009        47     11   10101         1     1    2    39    5   
...         ...       ...    ...     ...       ...   ...  ...   ...  ...   
28158     20692        84     11  150801         4     3    2    25    6   
28159     20692        84     11  150801         5     3    1    23    6   
28160     20692        99     11  150801         1     1    2    60    3   
28161     20692        99     11  150801         2     3    2    39    6   
28162     20692        99     11  150801         3     3    1    28    6   

       TICUEST01  
0              2  
1              2  
2              2  
3              2  
4              2  
...          ...  
28158          2  
28159          2  
28160          2  
28161          2  
28162          2  

[28163 rows x 10 columns]

## **Pregunta 3**

Cree las columnas ID_HOGAR e id_hogar en los data frames de los módulos 100 y 200, respectivamente, concatenando los datos de conglomerado, vivienda y hogar.

In [17]:
datos1 = datos1.assign(ID_HOGAR = datos1['CONGLOME'].astype(str) + datos1['VIVIENDA'].astype(str) + datos1['HOGAR'].astype(str))

In [18]:
datos1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11226 entries, 0 to 11225
Data columns (total 15 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   CONGLOME  11226 non-null  int64 
 1   VIVIENDA  11226 non-null  int64 
 2   HOGAR     11226 non-null  int64 
 3   UBIGEO    11226 non-null  int64 
 4   FECENT    11226 non-null  int64 
 5   P101      11226 non-null  object
 6   P104      11226 non-null  object
 7   P106      11226 non-null  object
 8   P106A     11226 non-null  object
 9   P110      11226 non-null  object
 10  P1141     11226 non-null  object
 11  P1142     11226 non-null  object
 12  P1143     11226 non-null  object
 13  P1144     11226 non-null  object
 14  ID_HOGAR  11226 non-null  object
dtypes: int64(5), object(10)
memory usage: 1.3+ MB


In [19]:
datos2 = datos2.assign(id_hogar = datos2['CONGLOME'].astype(str) + datos2['VIVIENDA'].astype(str) + datos2['HOGAR'].astype(str))

In [20]:
datos2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28163 entries, 0 to 28162
Data columns (total 11 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   CONGLOME   28163 non-null  int64 
 1   VIVIENDA   28163 non-null  int64 
 2   HOGAR      28163 non-null  int64 
 3   UBIGEO     28163 non-null  int64 
 4   CODPERSO   28163 non-null  int64 
 5   P203       28163 non-null  int64 
 6   P207       28163 non-null  object
 7   P208A      28163 non-null  object
 8   P209       28163 non-null  object
 9   TICUEST01  28163 non-null  int64 
 10  id_hogar   28163 non-null  object
dtypes: int64(7), object(4)
memory usage: 2.4+ MB


In [21]:
datos1

CONGLOME  VIVIENDA  HOGAR  UBIGEO    FECENT P101 P104  P106 P106A P110  \
0         15009        10     11   10101  20240123                              
1         15009        22     11   10101  20240124    1    6  3000     1    1   
2         15009        47     11   10101  20240130    1    5   750          1   
3         15009        59     11   10101  20240117                              
4         15009        71     11   10101  20240128    3    1   400          1   
...         ...       ...    ...     ...       ...  ...  ...   ...   ...  ...   
11221     20692        40     11  150801  20240330    1    4   100     2    6   
11222     20692        55     11  150801  20240329                              
11223     20692        70     11  150801  20240329    1    2    50     1    8   
11224     20692        84     11  150801  20240329    1    3   150     2    4   
11225     20692        99     11  150801  20240331    1    5   200     2    4   

      P1141 P1142 P1143 P1144   ID_HOGAR  
0                              150091011  
1         0     1     1     1  150092211  
2         1     1     1     1  150094711  
3                              150095911  
4         0     1     1     1  150097111  
...     ...   ...   ...   ...        ...  
11221     0     1     1     1  206924011  
11222                          206925511  
11223     0     1     1     1  206927011  
11224     0     1     1     1  206928411  
11225     0     1     1     1  206929911  

[11226 rows x 15 columns]

In [22]:
datos2

CONGLOME  VIVIENDA  HOGAR  UBIGEO  CODPERSO  P203 P207 P208A P209  \
0         15009        22     11   10101         1     1    1    52    2   
1         15009        22     11   10101         2     2    2    47    2   
2         15009        22     11   10101         3     3    1    18    6   
3         15009        22     11   10101         4     3    2    11        
4         15009        47     11   10101         1     1    2    39    5   
...         ...       ...    ...     ...       ...   ...  ...   ...  ...   
28158     20692        84     11  150801         4     3    2    25    6   
28159     20692        84     11  150801         5     3    1    23    6   
28160     20692        99     11  150801         1     1    2    60    3   
28161     20692        99     11  150801         2     3    2    39    6   
28162     20692        99     11  150801         3     3    1    28    6   

       TICUEST01   id_hogar  
0              2  150092211  
1              2  150092211  
2              2  150092211  
3              2  150092211  
4              2  150094711  
...          ...        ...  
28158          2  206928411  
28159          2  206928411  
28160          2  206929911  
28161          2  206929911  
28162          2  206929911  

[28163 rows x 11 columns]

## **Pregunta 4**

Usando las columnas ID_HOGAR e id_hogar, combine los data frames mediante las siguientes reglas:

  a) Nombre de data frame: datos_personas
  
  Considerar a todas las personas, independientemente si pertenecen o no a un hogar.

In [23]:
datos_personas = pd.merge(datos1, datos2, left_on='ID_HOGAR', right_on='id_hogar', how='right', indicator=True)
datos_personas

CONGLOME_x  VIVIENDA_x  HOGAR_x  UBIGEO_x    FECENT P101 P104  P106  \
0           15009          22       11     10101  20240124    1    6  3000   
1           15009          22       11     10101  20240124    1    6  3000   
2           15009          22       11     10101  20240124    1    6  3000   
3           15009          22       11     10101  20240124    1    6  3000   
4           15009          47       11     10101  20240130    1    5   750   
...           ...         ...      ...       ...       ...  ...  ...   ...   
28158       20692          84       11    150801  20240329    1    3   150   
28159       20692          84       11    150801  20240329    1    3   150   
28160       20692          99       11    150801  20240331    1    5   200   
28161       20692          99       11    150801  20240331    1    5   200   
28162       20692          99       11    150801  20240331    1    5   200   

      P106A P110 P1141 P1142 P1143 P1144   ID_HOGAR  CONGLOME_y  VIVIENDA_y  \
0         1    1     0     1     1     1  150092211       15009          22   
1         1    1     0     1     1     1  150092211       15009          22   
2         1    1     0     1     1     1  150092211       15009          22   
3         1    1     0     1     1     1  150092211       15009          22   
4              1     1     1     1     1  150094711       15009          47   
...     ...  ...   ...   ...   ...   ...        ...         ...         ...   
28158     2    4     0     1     1     1  206928411       20692          84   
28159     2    4     0     1     1     1  206928411       20692          84   
28160     2    4     0     1     1     1  206929911       20692          99   
28161     2    4     0     1     1     1  206929911       20692          99   
28162     2    4     0     1     1     1  206929911       20692          99   

       HOGAR_y  UBIGEO_y  CODPERSO  P203 P207 P208A P209  TICUEST01  \
0           11     10101         1     1    1    52    2          2   
1           11     10101         2     2    2    47    2          2   
2           11     10101         3     3    1    18    6          2   
3           11     10101         4     3    2    11               2   
4           11     10101         1     1    2    39    5          2   
...        ...       ...       ...   ...  ...   ...  ...        ...   
28158       11    150801         4     3    2    25    6          2   
28159       11    150801         5     3    1    23    6          2   
28160       11    150801         1     1    2    60    3          2   
28161       11    150801         2     3    2    39    6          2   
28162       11    150801         3     3    1    28    6          2   

        id_hogar _merge  
0      150092211   both  
1      150092211   both  
2      150092211   both  
3      150092211   both  
4      150094711   both  
...          ...    ...  
28158  206928411   both  
28159  206928411   both  
28160  206929911   both  
28161  206929911   both  
28162  206929911   both  

[28163 rows x 27 columns]

In [24]:
datos_personas[datos_personas['_merge'] != "both"] # filtramos los datos de modo que _merge sea distinto a both
# filtramos los datos que no hayan coincidido en ambas tablas

Empty DataFrame
Columns: [CONGLOME_x, VIVIENDA_x, HOGAR_x, UBIGEO_x, FECENT, P101, P104, P106, P106A, P110, P1141, P1142, P1143, P1144, ID_HOGAR, CONGLOME_y, VIVIENDA_y, HOGAR_y, UBIGEO_y, CODPERSO, P203, P207, P208A, P209, TICUEST01, id_hogar, _merge]
Index: []

  b) Nombre de data frame: datos_hogares
  
  Considerar a todos los hogares, independientemente si encuestaron a alguna persona o no dentro de cada uno de ellos.

In [25]:
datos_hogares = pd.merge(datos1, datos2, left_on='ID_HOGAR', right_on='id_hogar', how='left', indicator=True)
datos_hogares

CONGLOME_x  VIVIENDA_x  HOGAR_x  UBIGEO_x    FECENT P101 P104  P106  \
0           15009          10       11     10101  20240123                   
1           15009          22       11     10101  20240124    1    6  3000   
2           15009          22       11     10101  20240124    1    6  3000   
3           15009          22       11     10101  20240124    1    6  3000   
4           15009          22       11     10101  20240124    1    6  3000   
...           ...         ...      ...       ...       ...  ...  ...   ...   
31417       20692          84       11    150801  20240329    1    3   150   
31418       20692          84       11    150801  20240329    1    3   150   
31419       20692          99       11    150801  20240331    1    5   200   
31420       20692          99       11    150801  20240331    1    5   200   
31421       20692          99       11    150801  20240331    1    5   200   

      P106A P110 P1141 P1142 P1143 P1144   ID_HOGAR  CONGLOME_y  VIVIENDA_y  \
0                                         150091011         NaN         NaN   
1         1    1     0     1     1     1  150092211     15009.0        22.0   
2         1    1     0     1     1     1  150092211     15009.0        22.0   
3         1    1     0     1     1     1  150092211     15009.0        22.0   
4         1    1     0     1     1     1  150092211     15009.0        22.0   
...     ...  ...   ...   ...   ...   ...        ...         ...         ...   
31417     2    4     0     1     1     1  206928411     20692.0        84.0   
31418     2    4     0     1     1     1  206928411     20692.0        84.0   
31419     2    4     0     1     1     1  206929911     20692.0        99.0   
31420     2    4     0     1     1     1  206929911     20692.0        99.0   
31421     2    4     0     1     1     1  206929911     20692.0        99.0   

       HOGAR_y  UBIGEO_y  CODPERSO  P203 P207 P208A P209  TICUEST01  \
0          NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN   
1         11.0   10101.0       1.0   1.0    1    52    2        2.0   
2         11.0   10101.0       2.0   2.0    2    47    2        2.0   
3         11.0   10101.0       3.0   3.0    1    18    6        2.0   
4         11.0   10101.0       4.0   3.0    2    11             2.0   
...        ...       ...       ...   ...  ...   ...  ...        ...   
31417     11.0  150801.0       4.0   3.0    2    25    6        2.0   
31418     11.0  150801.0       5.0   3.0    1    23    6        2.0   
31419     11.0  150801.0       1.0   1.0    2    60    3        2.0   
31420     11.0  150801.0       2.0   3.0    2    39    6        2.0   
31421     11.0  150801.0       3.0   3.0    1    28    6        2.0   

        id_hogar     _merge  
0            NaN  left_only  
1      150092211       both  
2      150092211       both  
3      150092211       both  
4      150092211       both  
...          ...        ...  
31417  206928411       both  
31418  206928411       both  
31419  206929911       both  
31420  206929911       both  
31421  206929911       both  

[31422 rows x 27 columns]

In [26]:
datos_hogares[datos_hogares['_merge'] != "both"]

CONGLOME_x  VIVIENDA_x  HOGAR_x  UBIGEO_x    FECENT P101 P104 P106  \
0           15009          10       11     10101  20240123                  
8           15009          59       11     10101  20240117                  
24          15024          11       11     10101  20240116                  
30          15024          54       11     10101  20240116                  
36          15024          98       12     10101  20240117                  
...           ...         ...      ...       ...       ...  ...  ...  ...   
31363       20688          99       11    160201  20240309                  
31364       20688         114       11    160201  20240309                  
31369       20688         130       11    160201  20240309                  
31377       20689          54       11    200501  20240317                  
31408       20692          55       11    150801  20240329                  

      P106A P110 P1141 P1142 P1143 P1144    ID_HOGAR  CONGLOME_y  VIVIENDA_y  \
0                                          150091011         NaN         NaN   
8                                          150095911         NaN         NaN   
24                                         150241111         NaN         NaN   
30                                         150245411         NaN         NaN   
36                                         150249812         NaN         NaN   
...     ...  ...   ...   ...   ...   ...         ...         ...         ...   
31363                                      206889911         NaN         NaN   
31364                                     2068811411         NaN         NaN   
31369                                     2068813011         NaN         NaN   
31377                                      206895411         NaN         NaN   
31408                                      206925511         NaN         NaN   

       HOGAR_y  UBIGEO_y  CODPERSO  P203 P207 P208A P209  TICUEST01 id_hogar  \
0          NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   
8          NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   
24         NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   
30         NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   
36         NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   
...        ...       ...       ...   ...  ...   ...  ...        ...      ...   
31363      NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   
31364      NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   
31369      NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   
31377      NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   
31408      NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   

          _merge  
0      left_only  
8      left_only  
24     left_only  
30     left_only  
36     left_only  
...          ...  
31363  left_only  
31364  left_only  
31369  left_only  
31377  left_only  
31408  left_only  

[3259 rows x 27 columns]

  c) Nombre de data frame: datos_int
  
  Considerar a todos los hogares en los que haya sido entrevistada alguna persona, y en consecuencia a todas las personas que pertenezcan a algún hogar (no hogares vacíos ni personas sin vivienda u hogar).

In [27]:
datos_int = pd.merge(datos1, datos2, left_on='ID_HOGAR', right_on='id_hogar', how='inner', indicator=True)
datos_int

CONGLOME_x  VIVIENDA_x  HOGAR_x  UBIGEO_x    FECENT P101 P104  P106  \
0           15009          22       11     10101  20240124    1    6  3000   
1           15009          22       11     10101  20240124    1    6  3000   
2           15009          22       11     10101  20240124    1    6  3000   
3           15009          22       11     10101  20240124    1    6  3000   
4           15009          47       11     10101  20240130    1    5   750   
...           ...         ...      ...       ...       ...  ...  ...   ...   
28158       20692          84       11    150801  20240329    1    3   150   
28159       20692          84       11    150801  20240329    1    3   150   
28160       20692          99       11    150801  20240331    1    5   200   
28161       20692          99       11    150801  20240331    1    5   200   
28162       20692          99       11    150801  20240331    1    5   200   

      P106A P110 P1141 P1142 P1143 P1144   ID_HOGAR  CONGLOME_y  VIVIENDA_y  \
0         1    1     0     1     1     1  150092211       15009          22   
1         1    1     0     1     1     1  150092211       15009          22   
2         1    1     0     1     1     1  150092211       15009          22   
3         1    1     0     1     1     1  150092211       15009          22   
4              1     1     1     1     1  150094711       15009          47   
...     ...  ...   ...   ...   ...   ...        ...         ...         ...   
28158     2    4     0     1     1     1  206928411       20692          84   
28159     2    4     0     1     1     1  206928411       20692          84   
28160     2    4     0     1     1     1  206929911       20692          99   
28161     2    4     0     1     1     1  206929911       20692          99   
28162     2    4     0     1     1     1  206929911       20692          99   

       HOGAR_y  UBIGEO_y  CODPERSO  P203 P207 P208A P209  TICUEST01  \
0           11     10101         1     1    1    52    2          2   
1           11     10101         2     2    2    47    2          2   
2           11     10101         3     3    1    18    6          2   
3           11     10101         4     3    2    11               2   
4           11     10101         1     1    2    39    5          2   
...        ...       ...       ...   ...  ...   ...  ...        ...   
28158       11    150801         4     3    2    25    6          2   
28159       11    150801         5     3    1    23    6          2   
28160       11    150801         1     1    2    60    3          2   
28161       11    150801         2     3    2    39    6          2   
28162       11    150801         3     3    1    28    6          2   

        id_hogar _merge  
0      150092211   both  
1      150092211   both  
2      150092211   both  
3      150092211   both  
4      150094711   both  
...          ...    ...  
28158  206928411   both  
28159  206928411   both  
28160  206929911   both  
28161  206929911   both  
28162  206929911   both  

[28163 rows x 27 columns]

In [28]:
datos_int[datos_int['_merge'] != "both"]

Empty DataFrame
Columns: [CONGLOME_x, VIVIENDA_x, HOGAR_x, UBIGEO_x, FECENT, P101, P104, P106, P106A, P110, P1141, P1142, P1143, P1144, ID_HOGAR, CONGLOME_y, VIVIENDA_y, HOGAR_y, UBIGEO_y, CODPERSO, P203, P207, P208A, P209, TICUEST01, id_hogar, _merge]
Index: []

 d) Nombre de data frame: datos_uni

  Considerar a todos los hogares (con o sin personas entrevistadas) así como a todas las personas (que pertenezcan o no a algún hogar).

In [29]:
datos_uni = pd.merge(datos1, datos2, left_on='ID_HOGAR', right_on='id_hogar', how='outer',indicator=True)
datos_uni

CONGLOME_x  VIVIENDA_x  HOGAR_x  UBIGEO_x    FECENT P101 P104  P106  \
0           15009          10       11     10101  20240123                   
1           15009          22       11     10101  20240124    1    6  3000   
2           15009          22       11     10101  20240124    1    6  3000   
3           15009          22       11     10101  20240124    1    6  3000   
4           15009          22       11     10101  20240124    1    6  3000   
...           ...         ...      ...       ...       ...  ...  ...   ...   
31417       20692          84       11    150801  20240329    1    3   150   
31418       20692          84       11    150801  20240329    1    3   150   
31419       20692          99       11    150801  20240331    1    5   200   
31420       20692          99       11    150801  20240331    1    5   200   
31421       20692          99       11    150801  20240331    1    5   200   

      P106A P110 P1141 P1142 P1143 P1144   ID_HOGAR  CONGLOME_y  VIVIENDA_y  \
0                                         150091011         NaN         NaN   
1         1    1     0     1     1     1  150092211     15009.0        22.0   
2         1    1     0     1     1     1  150092211     15009.0        22.0   
3         1    1     0     1     1     1  150092211     15009.0        22.0   
4         1    1     0     1     1     1  150092211     15009.0        22.0   
...     ...  ...   ...   ...   ...   ...        ...         ...         ...   
31417     2    4     0     1     1     1  206928411     20692.0        84.0   
31418     2    4     0     1     1     1  206928411     20692.0        84.0   
31419     2    4     0     1     1     1  206929911     20692.0        99.0   
31420     2    4     0     1     1     1  206929911     20692.0        99.0   
31421     2    4     0     1     1     1  206929911     20692.0        99.0   

       HOGAR_y  UBIGEO_y  CODPERSO  P203 P207 P208A P209  TICUEST01  \
0          NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN   
1         11.0   10101.0       1.0   1.0    1    52    2        2.0   
2         11.0   10101.0       2.0   2.0    2    47    2        2.0   
3         11.0   10101.0       3.0   3.0    1    18    6        2.0   
4         11.0   10101.0       4.0   3.0    2    11             2.0   
...        ...       ...       ...   ...  ...   ...  ...        ...   
31417     11.0  150801.0       4.0   3.0    2    25    6        2.0   
31418     11.0  150801.0       5.0   3.0    1    23    6        2.0   
31419     11.0  150801.0       1.0   1.0    2    60    3        2.0   
31420     11.0  150801.0       2.0   3.0    2    39    6        2.0   
31421     11.0  150801.0       3.0   3.0    1    28    6        2.0   

        id_hogar     _merge  
0            NaN  left_only  
1      150092211       both  
2      150092211       both  
3      150092211       both  
4      150092211       both  
...          ...        ...  
31417  206928411       both  
31418  206928411       both  
31419  206929911       both  
31420  206929911       both  
31421  206929911       both  

[31422 rows x 27 columns]

In [30]:
datos_uni[datos_uni['_merge'] != "both"]

CONGLOME_x  VIVIENDA_x  HOGAR_x  UBIGEO_x    FECENT P101 P104 P106  \
0           15009          10       11     10101  20240123                  
8           15009          59       11     10101  20240117                  
24          15024          11       11     10101  20240116                  
30          15024          54       11     10101  20240116                  
36          15024          98       12     10101  20240117                  
...           ...         ...      ...       ...       ...  ...  ...  ...   
31363       20688          99       11    160201  20240309                  
31364       20688         114       11    160201  20240309                  
31369       20688         130       11    160201  20240309                  
31377       20689          54       11    200501  20240317                  
31408       20692          55       11    150801  20240329                  

      P106A P110 P1141 P1142 P1143 P1144    ID_HOGAR  CONGLOME_y  VIVIENDA_y  \
0                                          150091011         NaN         NaN   
8                                          150095911         NaN         NaN   
24                                         150241111         NaN         NaN   
30                                         150245411         NaN         NaN   
36                                         150249812         NaN         NaN   
...     ...  ...   ...   ...   ...   ...         ...         ...         ...   
31363                                      206889911         NaN         NaN   
31364                                     2068811411         NaN         NaN   
31369                                     2068813011         NaN         NaN   
31377                                      206895411         NaN         NaN   
31408                                      206925511         NaN         NaN   

       HOGAR_y  UBIGEO_y  CODPERSO  P203 P207 P208A P209  TICUEST01 id_hogar  \
0          NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   
8          NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   
24         NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   
30         NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   
36         NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   
...        ...       ...       ...   ...  ...   ...  ...        ...      ...   
31363      NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   
31364      NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   
31369      NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   
31377      NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   
31408      NaN       NaN       NaN   NaN  NaN   NaN  NaN        NaN      NaN   

          _merge  
0      left_only  
8      left_only  
24     left_only  
30     left_only  
36     left_only  
...          ...  
31363  left_only  
31364  left_only  
31369  left_only  
31377  left_only  
31408  left_only  

[3259 rows x 27 columns]

## **Pregunta 5**

A partir de esta pregunta, ejecutar los códigos usando datos_int. Cambiar los nombres de las variables:

  - CONGLOME_x: Conglome
  -	VIVIENDA_x: Vivienda
  - HOGAR_x: Hogar
  - UBIGEO_x: Ubigeo
  - FECENT: FecEnt
  - P101: TipoVivienda
  - P104: NumHab
  - P106: ValorAlq
  - P106A: Titulo
  - P110: Agua
  - P1141: TelFijo
  - P1142: TelCel
  - P1143: tvCable
  - P1144: Internet
  - CODPERSO: CodPerso
  - P203: Parentesco
  - P207: Sexo
  - P208A: Edad
  - P209: EstCivil
  - TICUEST01: TipoCuest

In [33]:
# prompt: Show me an example of a dictionary

my_dict = {"apple": 1, "banana": 2, "cherry": 3}
my_dict['banana']

2

In [34]:
datos_int.head(4)

CONGLOME_x  VIVIENDA_x  HOGAR_x  UBIGEO_x    FECENT P101 P104  P106 P106A  \
0       15009          22       11     10101  20240124    1    6  3000     1   
1       15009          22       11     10101  20240124    1    6  3000     1   
2       15009          22       11     10101  20240124    1    6  3000     1   
3       15009          22       11     10101  20240124    1    6  3000     1   

  P110 P1141 P1142 P1143 P1144   ID_HOGAR  CONGLOME_y  VIVIENDA_y  HOGAR_y  \
0    1     0     1     1     1  150092211       15009          22       11   
1    1     0     1     1     1  150092211       15009          22       11   
2    1     0     1     1     1  150092211       15009          22       11   
3    1     0     1     1     1  150092211       15009          22       11   

   UBIGEO_y  CODPERSO  P203 P207 P208A P209  TICUEST01   id_hogar _merge  
0     10101         1     1    1    52    2          2  150092211   both  
1     10101         2     2    2    47    2          2  150092211   both  
2     10101         3     3    1    18    6          2  150092211   both  
3     10101         4     3    2    11               2  150092211   both

In [35]:
nuevos_nombres = {'CONGLOME_x':'Conglome', 'VIVIENDA_x':'Vivienda', 'HOGAR_x':'Hogar', 'UBIGEO_x':'Ubigeo', 'FECENT':'FecEnt',
                  'P101':'TipoVivienda', 'P104':'NumHab', 'P106':'ValorAlq', 'P106A':'Titulo','P110':'Agua',
                  'P1141':'TelFijo', 'P1142':'TelCel', 'P1143':'tvCable', 'P1144':'Internet',
                  'CODPERSO':'CodPerso', 'P203':'Parentesco', 'P207':'Sexo', 'P208A':'Edad', 'P209':'EstCivil', 'TICUEST01': 'TipoCuest'}
datos_int.rename(columns=nuevos_nombres, inplace=True)
datos_int.head(4)

Conglome  Vivienda  Hogar  Ubigeo    FecEnt TipoVivienda NumHab ValorAlq  \
0     15009        22     11   10101  20240124            1      6     3000   
1     15009        22     11   10101  20240124            1      6     3000   
2     15009        22     11   10101  20240124            1      6     3000   
3     15009        22     11   10101  20240124            1      6     3000   

  Titulo Agua TelFijo TelCel tvCable Internet   ID_HOGAR  CONGLOME_y  \
0      1    1       0      1       1        1  150092211       15009   
1      1    1       0      1       1        1  150092211       15009   
2      1    1       0      1       1        1  150092211       15009   
3      1    1       0      1       1        1  150092211       15009   

   VIVIENDA_y  HOGAR_y  UBIGEO_y  CodPerso  Parentesco Sexo Edad EstCivil  \
0          22       11     10101         1           1    1   52        2   
1          22       11     10101         2           2    2   47        2   
2          22       11     10101         3           3    1   18        6   
3          22       11     10101         4           3    2   11            

   TipoCuest   id_hogar _merge  
0          2  150092211   both  
1          2  150092211   both  
2          2  150092211   both  
3          2  150092211   both

In [38]:
columnas_drop = [col for col in datos_int.columns if col.endswith("_y")] + ["id_hogar", "_merge"] # almacenamos en una lista las variables que serán eliminadas
datos_int.drop(columnas_drop, axis=1, inplace = True) # realizamos la eliminación de las variables que han sido definidas
datos_int.head(4)

Conglome  Vivienda  Hogar  Ubigeo    FecEnt TipoVivienda NumHab ValorAlq  \
0     15009        22     11   10101  20240124            1      6     3000   
1     15009        22     11   10101  20240124            1      6     3000   
2     15009        22     11   10101  20240124            1      6     3000   
3     15009        22     11   10101  20240124            1      6     3000   

  Titulo Agua TelFijo TelCel tvCable Internet   ID_HOGAR  CodPerso  \
0      1    1       0      1       1        1  150092211         1   
1      1    1       0      1       1        1  150092211         2   
2      1    1       0      1       1        1  150092211         3   
3      1    1       0      1       1        1  150092211         4   

   Parentesco Sexo Edad EstCivil  TipoCuest  
0           1    1   52        2          2  
1           2    2   47        2          2  
2           3    1   18        6          2  
3           3    2   11                   2

In [45]:
columnas = datos_int.columns.tolist() # Lista de nombres de variables (columnas)
columnas = ['ID_HOGAR'] + [col for col in columnas if col != 'ID_HOGAR']
datos_int = datos_int[columnas]

In [46]:
datos_int.head(4)

ID_HOGAR  Conglome  Vivienda  Hogar  Ubigeo    FecEnt TipoVivienda NumHab  \
0  150092211     15009        22     11   10101  20240124            1      6   
1  150092211     15009        22     11   10101  20240124            1      6   
2  150092211     15009        22     11   10101  20240124            1      6   
3  150092211     15009        22     11   10101  20240124            1      6   

  ValorAlq Titulo Agua TelFijo TelCel tvCable Internet  CodPerso  Parentesco  \
0     3000      1    1       0      1       1        1         1           1   
1     3000      1    1       0      1       1        1         2           2   
2     3000      1    1       0      1       1        1         3           3   
3     3000      1    1       0      1       1        1         4           3   

  Sexo Edad EstCivil  TipoCuest  
0    1   52        2          2  
1    2   47        2          2  
2    1   18        6          2  
3    2   11                   2

In [42]:
# prompt: cómo mostrar el resultado de datos_int.columns.tolist() de manera horizontal?

# Usamos el método join para unir los elementos de la lista con un espacio en blanco
print(', '.join(datos_int.columns.tolist()))


Conglome, Vivienda, Hogar, Ubigeo, FecEnt, TipoVivienda, NumHab, ValorAlq, Titulo, Agua, TelFijo, TelCel, tvCable, Internet, ID_HOGAR, CodPerso, Parentesco, Sexo, Edad, EstCivil, TipoCuest


## **Pregunta 7**

Identificar cuantos valores perdidos hay en cada variable, ¿existe alguna totalmente completa? ¿cuál es la que tiene mayor cantidad de perdidos?

In [ ]:
datos_int.isna().sum(axis=0) # Suma la cantidad de datos perdidos por columnas

In [ ]:
datos_int.info()

In [ ]:
datos_int

In [ ]:
import numpy as np
datos_int.replace([" ", "","999","99999"], np.NAN, inplace=True)
datos_int.isna().sum(axis=0)

In [ ]:
datos_int

## **Pregunta 8**

Verifique que cada una de las variables sea del tipo que se indica a continuación. De no ser así, modificar:

- Conglome: string

- Vivienda: string

- Hogar: string

- Ubigeo: string

- FecEnt: string

- TipoVivienda: string

- NumHab: int

- ValorAlq: float

- Titulo: string

- Agua: string

- TelFijo: string

- TelCel: string

- tvCable: string

- Internet: string

- Parentesco: string

- Sexo: string

- Edad: int

- EstCivil: string

- TipoCuest: string

Revisar [aquí](https://pbpython.com/pandas_dtypes.html) acerca de los tipos de datos en Python

Ya que el resultado es False, el tipo de dato en Conglome no es string

In [ ]:
datos_int["Conglome"].dtype == str # se valida si el tipo de dato almacenado en Conglome es string

In [ ]:
datos_int["Conglome"]

In [ ]:
datos_int.dtypes

In [ ]:
datos_int["Conglome"] = datos_int["Conglome"].astype(str)

In [ ]:
for col in ["Vivienda", "Hogar", "Ubigeo", "FecEnt", "Parentesco", "TipoCuest"]:
  datos_int[col] = datos_int[col].apply(str)

In [ ]:
for col in ["NumHab", "Edad"]:
  datos_int[col] = datos_int[col].fillna(999)
  datos_int[col] = datos_int[col].apply(int)
  datos_int[col] = datos_int[col].replace(999, np.nan)

In [ ]:
for col in ["ValorAlq"]:
  datos_int[col] = datos_int[col].apply(float)

In [ ]:
datos_int.dtypes

## **Pregunta 9**

Recategorizar la variable TipoVivienda, de modo que los nuevos valores sean:

- Antes: Casa independiente (1), Departamento en edificio (2)

  Ahora: Vivienda adecuada (1)

- Antes: Vivienda en quinta (3), Vivienda en casa de vecindad (4), Choza o cabaña (5), Vivienda improvisada (6), Local no destinado para habitación humana (7)

  Ahora: Vivienda precaria (2)




In [ ]:
datos_int["TipoVivienda"].value_counts() # Obtenemos una tabla de frecuencias para la variable TipoVivienda, ordenada por frecuencias.

In [ ]:
datos_int["TipoVivienda"].value_counts().reset_index().sort_values(by = "TipoVivienda") # Obtenemos una tabla de frecuencias para la variable TipoVivienda, ordenada por categorías.

In [ ]:
datos_int.groupby('TipoVivienda').size().reset_index(name='Frecuencia') # Otra manera de obtener la tabla de frecuencias

In [ ]:
pip install researchpy

In [ ]:
import researchpy as rp

In [ ]:
rp.summary_cat(datos_int['TipoVivienda'])

In [ ]:
rp.summary_cat(datos_int['TipoVivienda'])

In [ ]:
datos_int["TipoVivienda2"] = datos_int["TipoVivienda"].replace({"1": "1",
                                                                "2": "1",
                                                                "3": "2",
                                                                "4": "2",
                                                                "5": "2",
                                                                "6": "2",
                                                                "7": "2"})
datos_int.groupby('TipoVivienda2').size().reset_index(name='Frecuencia')

In [ ]:
datos_int["TipoVivienda3"] = datos_int["TipoVivienda"].replace(["1","2"], "1")
datos_int["TipoVivienda3"] = datos_int["TipoVivienda3"].replace(["3","4","5","6","7"], "2")

In [ ]:
#datos_int["TipoVivienda"].replace(["1","2"], "1", inplace = True)
#datos_int["TipoVivienda"].replace(["3","4","5","6","7"], "2", inplace = True)
#¿Qué sucedería su ejecutamos este código?

In [ ]:
pd.crosstab(datos_int["TipoVivienda"], datos_int["TipoVivienda2"]) # Realizamos una tabla cruzada de la variable original y la variable recategorizada

In [ ]:
pd.crosstab(datos_int["TipoVivienda"], datos_int["TipoVivienda3"]) # Realizamos una tabla cruzada de la variable original y la variable recategorizada

De esta manera, hemos verificado que las categorías se han actualizado correctamente.

## **Pregunta 10**

Categorizar la variable Edad, de modo que se consideren las siguientes reglas:

- De 0 a 11 años: Niño
- De 12 a 17 años: Adolescente
- De 18 a 29 años: Joven
- De 30 a 59 años: Adulto
- De 60 a más años: Adulto mayor


In [ ]:
datos_int.Edad.value_counts()

In [ ]:
#datos_int.Edad.value_counts().reset_index().sort_values(by = "Edad")
datos_int.groupby('Edad').size().reset_index(name='Frecuencia')

In [ ]:
datos_int['Grupo_Edad'] = pd.cut(datos_int['Edad'],
                                 bins = [-1, 11, 17, 29, 59, 100, float('inf')],
                                 #labels = ['Niño', 'Adolescente', 'Joven', 'Adulto', 'Adulto mayor', 'Sin datos'],
                                 right = True)

In [ ]:
datos_int['Grupo_Edad'] = pd.cut(datos_int['Edad'],
                                 bins = [0, 12, 18, 30, 60, 101, float('inf')],
                                 labels = ['Niño', 'Adolescente', 'Joven', 'Adulto', 'Adulto mayor', 'Sin datos'],
                                 right = False)

In [ ]:
datos_int[['Edad','Grupo_Edad']]

In [ ]:
#datos_int.Grupo_Edad.value_counts().reset_index().sort_values(by = "Grupo_Edad")
datos_int.groupby('Grupo_Edad').size().reset_index(name='Frecuencia')

## **Pregunta 11**

Crear variables dummy a partir de la variable categórica creada en la pregunta anterior.

In [ ]:
datos_int

In [ ]:
datos_int = pd.get_dummies(datos_int, columns=['Grupo_Edad'], prefix=['DU'])
datos_int

In [ ]:
variables_dummy = pd.get_dummies(datos_int['EstCivil'].astype(str), prefix='EC')
datos_int = pd.concat([datos_int, variables_dummy], axis=1)
datos_int

### **Pregunta 12**

Estandarizar la variable ValorAlq

In [ ]:
datos_int['ValorAlq_st'] = (datos_int['ValorAlq'] - datos_int['ValorAlq'].mean()) / datos_int['ValorAlq'].std()
datos_int[['ValorAlq','ValorAlq_st']]

In [ ]:
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()
datos_int['ValorAlq_st2'] = scaler.fit_transform(datos_int[['ValorAlq']])
datos_int[['ValorAlq','ValorAlq_st','ValorAlq_st2']]

In [ ]:
datos_int['ValorAlq_st3'] = (datos_int['ValorAlq'] - datos_int['ValorAlq'].min()) / (datos_int['ValorAlq'].max()- datos_int['ValorAlq'].min())
datos_int[['ValorAlq','ValorAlq_st3']]

In [ ]:
from sklearn.preprocessing import MinMaxScaler
scaler = MinMaxScaler()
datos_int['ValorAlq_st4'] = scaler.fit_transform(datos_int[['ValorAlq']])
datos_int[['ValorAlq','ValorAlq_st3','ValorAlq_st4']]

### **Pregunta 13**

A partir de la columna FecEnt, crear una variable llamada Mes_Ent, que viene a ser el mes del resultado de la encuesta

In [ ]:
datos_int['FecEnt']

In [ ]:
datos_int['Mes_Ent']   = pd.to_datetime(datos_int['FecEnt']).dt.month

In [ ]:
datos_int[['FecEnt','Mes_Ent']]

In [ ]:
datos_int['Fecha_Ent'] = pd.to_datetime(datos_int['FecEnt'])
datos_int['Mes_Ent']   = datos_int['Fecha_Ent'].dt.month

datos_int['Mes_Ent']   = pd.to_datetime(datos_int['FecEnt']).dt.month

datos_int[['FecEnt','Fecha_Ent','Mes_Ent']]

¿Cuál fue el primer y último día en el que se ejecutó la encuesta correspondiente al primer trimestre del 2022?

In [ ]:
print('Primer día:', datos_int['Fecha_Ent'].min())

In [ ]:
print('Primer día:', datos_int['Fecha_Ent'].min().strftime('%Y-%m-%d'))

In [ ]:
print('Último día:', datos_int['Fecha_Ent'].max().strftime('%d-%b-%y'))

### **Pregunta 14**

Seleccionar a las personas que son menores de edad

In [ ]:
datos_f1 = datos_int.loc[datos_int['Edad'] < 18]
datos_f1[['Edad']].describe()

- Hay 8874 menores de edad en la muestra
- La edad promedio de los menores de edad es de 9.16 años
- Al menos el 50% de los menores de edad tiene 9 años como máximo.
- Al menos el 75% de los menores de edad tiene hasta 14 años.

### **Pregunta 15**

Seleccionar los hogares ubicados en viviendas que son adecuadas, que cuentan con teléfono fijo pero no internet.

¿La edad promedio de las personas de estos hogares es mayor que en aquellos que tienen estas mismas características excepto que no cuentan con teléfono fijo?

In [ ]:
# viviendas que son adecuadas, que cuentan con teléfono fijo pero no internet.
datos_f2 = datos_int.loc[(datos_int['TipoVivienda2'] == "1") & (datos_int['Internet'] == "0") & (datos_int['TelFijo'] == "1")]

In [ ]:
# viviendas que son adecuadas, que no cuentan con teléfono fijo pero no internet.
datos_f3 = datos_int.loc[(datos_int['TipoVivienda2'] == "1") & (datos_int['Internet'] == "0") & (datos_int['TelFijo'] == "0")]

In [ ]:
datos_f2.Edad.mean() # con teléfono fijo, sin internet

In [ ]:
datos_f3.Edad.mean() # sin teléfono fijo, sin internet

La edad promedio de las personas que habitan una vivienda adecuada, no tienen internet pero sí teléfono fijo es de 60.9 años. Mientras que aquellas que **no tienen el servicio de teléfono fijo**, tienen una edad media **26 años menor**.

### **Pregunta 16**

Filtrar a las mujeres que tienen más de 55 años y que habitan en viviendas con menos de 5 habitaciones y cuyo servicio de agua proviene de la red pública, dentro de la vivienda.

In [ ]:
# Importante: colocar cada condición entre paréntesis, de lo contrario arroja error
datos_f4 = datos_int.loc[(datos_int['Sexo'] == "2") & (datos_int['Edad']>55) & (datos_int['NumHab'] < 5) & (datos_int['Agua'] == "1")]
datos_f4[['Sexo','Edad','NumHab','Agua']]

In [ ]:
datos_f4[['Sexo','Edad','NumHab','Agua']].describe()

In [ ]:
datos_f4[['Sexo','Edad','NumHab','Agua']].describe(include = 'all')

### **Pregunta 17**

Seleccionar los hogares que cuentan con título de propiedad y que reciben agua de Pilón o pileta de uso público o Camión-cisterna u otro similar. Luego, verifique si el precio mediano de alquiler de estas viviendas es mayor a 350 soles, y comente su distribución / forma / asimetría.

In [ ]:
datos_f5 = datos_int.loc[(datos_int['Titulo'] == "1") & ((datos_int['Agua'] == "3") | (datos_int['Agua'] == "4"))]

In [ ]:
datos_f5.ValorAlq.median()

Al menos el 50% de los hogares que cuentan con título de propiedad y que reciben agua de Pilón o pileta de uso público o Camión-cisterna u otro similar considera que recibiría 150 soles o menos por alquiler

In [ ]:
from scipy.stats import skew
skew(datos_f5['ValorAlq'])

Los montos percibidos de alquiler tienen una distribución asimétrica a la derecha, eso significa que gran parte de los precios están concentrados en valores bajos.

In [ ]:
import matplotlib.pyplot as plt
plt.hist(datos_f5['ValorAlq'], color='skyblue', edgecolor='black')
plt.xlabel('Valor en soles')
plt.ylabel('Frecuencia')
plt.title('Distribución del precio')
plt.show()